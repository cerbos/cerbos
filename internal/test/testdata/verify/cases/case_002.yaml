description: Valid suite with inline fixtures mixed with external ones
