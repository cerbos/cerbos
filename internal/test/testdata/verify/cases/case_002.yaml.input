-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design
  bev: &bev
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr: &bev_attr
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design
  matt:
    << : *bev
    id: matt
    attr:
      << : *bev_attr
      ip_address: 10.10.1.2

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
  pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      << : *attr
      status: PENDING_APPROVAL
  stale_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- inline_test.yaml --
---
name: TestSuite
description: Tests for verifying something
principals:
  johnny:
    id: johnny
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design
auxData:
  johnJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]
tests:
  - name: John and his leave request
    input:
      actions:
        - view:public
        - approve
        - defer
      principals:
        - johnny
      resources:
        - john_leave_request
      auxData: johnJWT
    expected:
      - principal: johnny
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW
          approve: EFFECT_DENY
          defer: EFFECT_ALLOW
