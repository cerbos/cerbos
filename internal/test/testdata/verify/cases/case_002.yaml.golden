{
  "suites": [
    {
      "file": "inline_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 3,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 3
          }
        ]
      },
      "testCases": [
        {
          "name": "John and his leave request",
          "principals": [
            {
              "name": "johnny",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    },
                    {
                      "name": "defer",
                      "details": {
                        "result": "RESULT_PASSED"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 3,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 3
      }
    ]
  }
}